container-deployment/docker-compose.md

```diff
@@ -0,0 +1,207 @@
+forwarding logs to Elasticsearch.
+
+``` {.CodeRay}
+# fluentd/conf/fluent.conf
+<source>
+  @type forward
+  port 24224
+  bind 0.0.0.0
+</source>
+<match *.**>
+  @type copy
+  <store>
+    @type elasticsearch
+    host elasticsearch
+    port 9200
+    logstash_format true
+    logstash_prefix fluentd
+    logstash_dateformat %Y%m%d
+    include_tag_key true
+    type_name access_log
+    tag_key @log_name
+    flush_interval 1s
+  </store>
+  <store>
+    @type stdout
+  </store>
+</match>
+```
+
+## Step 2: Start Containers
+
+Let's start all of the containers, with just one command.
+
+``` {.CodeRay}
+$ docker-compose up
+```
+
+You can check to see if 4 containers are running by `docker ps` command.
+
+``` {.CodeRay}
+$ docker ps
+CONTAINER ID        IMAGE                                                 COMMAND                  CREATED             STATUS              PORTS                              NAMES
+558fd18fa2d4        httpd                                                 "httpd-foreground"       17 seconds ago      Up 16 seconds       0.0.0.0:80->80/tcp                 docker_web_1
+bc5bcaedb282        kibana:7.2.0                                          "/usr/local/bin/kiba…"   18 seconds ago      Up 17 seconds       0.0.0.0:5601->5601/tcp             docker_kibana_1
+9fe2d02cff41        docker.elastic.co/elasticsearch/elasticsearch:7.2.0   "/usr/local/bin/dock…"   20 seconds ago      Up 18 seconds       0.0.0.0:9200->9200/tcp, 9300/tcp   docker_elasticsearch_1
+```
+
+## Step 3: Generate httpd Access Logs
+
+Let's access to `httpd` to generate some access logs. `curl` command is
+always your friend.
+
+``` {.CodeRay}
+$ curl http://localhost:80/[1-10]
+<html><body><h1>It works!</h1></body></html>
+<html><body><h1>It works!</h1></body></html>
+<html><body><h1>It works!</h1></body></html>
+<html><body><h1>It works!</h1></body></html>
+<html><body><h1>It works!</h1></body></html>
+<html><body><h1>It works!</h1></body></html>
+<html><body><h1>It works!</h1></body></html>
+<html><body><h1>It works!</h1></body></html>
+<html><body><h1>It works!</h1></body></html>
+<html><body><h1>It works!</h1></body></html>
+```
+
+## Step 4: Confirm Logs from Kibana
+
+Please go to `http://localhost:5601/` with your browser. Then, you need
+to set up the index name pattern for Kibana. Please specify `fluentd-*`
+to `Index name or pattern` and press `Create` button.
+
+![](/images/7.2_efk-kibana-index.png)
+![](/images/7.2_efk-kibana-timestamp.png)
+
+Then, go to `Discover` tab to seek for the logs. As you can see, logs
+are properly collected into Elasticsearch + Kibana, via Fluentd.
+
+![](/images/7.2_efk-kibana-discover.png)
+
+## Conclusion
+
+This article explains how to collect logs from Apache to EFK
+(Elasticsearch + Fluentd + Kibana). The example code is available in
+this repository.
+-   <https://github.com/digikin/fluentd-elastic-kibana>
+-   <https://github.com/kzk/docker-compose-efk>
```

Comment:
could you remove this line?

Reply:
Are you wanting me to remove line 194 and 195? I made a repo that contained all the example code so it was easy to pull the example?

Reply:
No. Only L195. L195 is not related with this article.